### DISCUSSION TITLE
Interface to see order of parameters in LogPDFs

### DISCUSSION THREAD

**Discussion by DavAug** [pints-team/pints#1649] [Ideas/Science]
I was wondering whether it may be beneficial to introduce an optional pints.LogPDF.parameter() method that returns, an array-like object with the names of the parameters. Maybe accompanied with optional pints.SingleOutputProblem.parameter() and pints.MultiOutputProblem.parameter().

While the the order of parameters is quite straightforward in case of a log-likelihood that only has one parameter it gets more confusing when the log-likelihood has multiple parameters. Especially when we have a multi-output problem the order of parameters is not obvious.

For example in pints.MultiplicativeGaussianLogLikelihood the model parameters `eta` and `sigma` are appended alternatingly for multiple outputs of the model. The parameter() method could then just return the names of all parameters to avoid confusion. If the underlying problems have no specified names, one could just return model_parameter_1, model_parameter_2, etc.. The important bit would be that the likelihoods could specify the order of their parameters.

**Comment by MichaelClerx** [comment]
See https://github.com/pints-team/pints/issues/816
And to a lesser degree https://github.com/pints-team/pints/issues/1072

**Comment by MichaelClerx** [comment]
Might be best to just do a proposal and see where/if it goes wrong
